backend/server/controllers/posts.js

```diff
@@ -0,0 +1,152 @@
+import models from '../../database/models'
+import { Authentication } from '../../middlewares/authentication'
+
+/*
+Requires authentication.
+Response codes:
+201 CREATED when the post has successfully been created.
+500 INTERNAL SERVER ERROR otherwise.
+*/
+export const createPost = async (req, res) => {
+    try {
+        const authToken = req.get('Authorization')
+
+        if (!authToken) {
+            res.status(400).send({
+                'Error message': 'Auth token not provided',
+            })
+        }
+
+        const decodedUser = Authentication.extractUser(authToken)
+
+        if (!decodedUser.id) {
+            res.status(401).send({
+                'Error message': 'Auth token invalid',
+            })
+        } else {
+            const { body } = req
+            /*
+            NOTE: attachments is current a STRING type as the attachment functionality has not been setup yet.
+            */
+            const createNewPost = await models.posts.create({
+                text_content: body.text_content,
+                author: 'test author',
```

Comment:
I think this line should be "body.author" not "test author" :satisfied:

Reply:
I changed it to author: decodedUser.username
I'm not sure if the author is suppose to be the author's id or the author's username though.